rdmo/management/tests/test_frontend.py

```diff
@@ -0,0 +1,210 @@
+
+    url: str
+    model: Model
+    form_field: str = "URI Path"
+    db_field: str = "uri_path"
+    has_nested: bool = False
+
+    @property
+    def url_name(self) -> str:
+        """Return the url name for the model."""
+        url = self.url.rstrip("s")
+        url = url.replace("set", " set")
+        return url
+
+
+@pytest.fixture(scope="function")
+def e2e_tests_django_db_setup(django_db_setup, django_db_blocker, fixtures):
+    """Set up database and populate with fixtures, that get restored for every test case."""
+    with django_db_blocker.unblock():
+        call_command("loaddata", *fixtures)
+        set_group_permissions()
+
+
+@pytest.fixture(scope="session")
+def base_url(live_server: LiveServer) -> str:
+    """Enable playwright to address URLs with base URL automatically prefixed."""
+    return live_server.url
+
+
+@pytest.fixture
+def logged_in_admin_user(e2e_tests_django_db_setup, page: Page) -> Page:
+    """Log in as admin user through django login UI, returns logged in page for e2e tests."""
+    page.goto("/account/login")
+    page.get_by_label("Username").fill("admin", timeout=5000)
+    page.get_by_label("Password").fill("admin")
+    page.get_by_role("button", name="Login").click()
+    page.goto("/management")
+    yield page
+
+
+def test_management_navigation(logged_in_admin_user: Page) -> None:
+    """Test that each content type is available through the navigation."""
+    page = logged_in_admin_user
+    expect(page.get_by_role("heading", name="Management")).to_be_visible()
+    page.get_by_role("link", name="Catalogs").click()
+    expect(page).to_have_url(re.compile(r".*/catalogs/"))
+    expect(page.locator("strong").filter(has_text="Catalogs")).to_be_visible()
+    page.get_by_role("link", name="Sections").click()
+    expect(page).to_have_url(re.compile(r".*/sections/"))
+    page.get_by_role("link", name="Pages").click()
+    expect(page).to_have_url(re.compile(r".*/pages/"))
+    page.get_by_role("link", name="Question sets").click()
+    expect(page).to_have_url(re.compile(r".*/questionsets/"))
+    page.get_by_role("link", name="Questions", exact=True).click()
+    expect(page).to_have_url(re.compile(r".*/questions/"))
+    page.get_by_role("link", name="Attributes").click()
+    expect(page).to_have_url(re.compile(r".*/attributes/"))
+    page.get_by_role("link", name="Option sets").click()
+    expect(page).to_have_url(re.compile(r".*/optionsets/"))
+    page.get_by_role("link", name="Options", exact=True).click()
+    expect(page).to_have_url(re.compile(r".*/options/"))
+    page.get_by_role("link", name="Conditions").click()
+    expect(page).to_have_url(re.compile(r".*/conditions/"))
+    page.get_by_role("link", name="Tasks").click()
+    expect(page).to_have_url(re.compile(r".*/tasks/"))
+    page.get_by_role("link", name="Views").click()
+    expect(page).to_have_url(re.compile(r".*/views/"))
+
+
+nested_elements = (
```

Comment:
this `nested_elements` is not used anymore, it can be removed now.
The propery is included in the `model_helpers` , `has_nested` kwarg.

Reply:
I see, thank you. Will remove it then.

Do you think the ModelHelper is a good idea for this? I just took a dataclass to have some information at hand and not to have so much duplication in the test functions.

Reply:
Done.

Reply:
Alright, I can add these methods to the helper.

Reply:
I added the properties. Where do you need verbose name and plural?